Foundation/NSArray.swift

```diff
@@ -171,44 +151,46 @@ open class NSArray : NSObject, NSCopying, NSMutableCopying, NSSecureCoding, NSCo
             return false
         }
         let otherArray = otherObject as! NSArray
-        return self.isEqual(to: otherArray.bridge())
+        return self.isEqual(to: otherArray.allObjects)
     }
 
     open override var hash: Int {
         return self.count
     }
 
-    internal var allObjects: [AnyObject] {
+    internal var allObjects: [Any] {
         if type(of: self) === NSArray.self || type(of: self) === NSMutableArray.self {
-            return _storage
+            return _storage.map { _SwiftValue.fetch($0) }
         } else {
             return (0..<count).map { idx in
                 return self[idx]
             }
         }
     }
     
-    open func adding(_ anObject: AnyObject) -> [AnyObject] {
+    open func adding(_ anObject: Any) -> [Any] {
         return allObjects + [anObject]
     }
     
-    open func addingObjects(from otherArray: [AnyObject]) -> [AnyObject] {
+    open func addingObjects(from otherArray: [Any]) -> [Any] {
         return allObjects + otherArray
     }
     
     open func componentsJoined(by separator: String) -> String {
         // make certain to call NSObject's description rather than asking the string interpolator for the swift description
-        return bridge().map() { ($0 as! NSObject).description }.joined(separator: separator)
+        return allObjects.map { "\($0)" }.joined(separator: separator)
     }
 
-    open func contains(_ anObject: AnyObject) -> Bool {
-        let other = anObject as! NSObject
+    open func contains(_ anObject: Any) -> Bool {
+        guard let other = anObject as? AnyHashable else {
```

Comment:
There is currently no facility from the language to do that because you would need to reach into the protocol and grab the `==` function to compare two items with `Self` types which has been destroyed by this point. the only way to obtain a clue approximation is to use `AnyHashable`.